pkg/controller/catalogsource/catalogsource_controller.go

```diff
@@ -87,12 +89,12 @@ type ReconcileCatalogSource struct {
 	client client.Client
 }
 
-func (r *ReconcileCatalogSource) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+func (r *ReconcileCatalogSource) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
```

Comment:
Is it necessary/helpful/reasonable to pass this context down to anything called from this function (e.g. `GetSingleton().Get()` or `Ensure()`)?

Reply:
I did a cursory look through the repository and removed as many usages of `context.TODO()` outside of testing code that were easily replaceable. The status reported package, which is responsible for setting up a go routine to monitor the marketplace ClusterOperator resource (and as aside, I think we'd want to move that handlement to its own controller down the line) but the changeset for updating all the call sites to properly handle signals seemed riskier in the full context of these changes, so I held off there.

One other quick note, and something we can tackle outside of this PR, is we still embed OperatorSource logic in the various top-level functions/methods/etc. I wonder if it's worth the easier maintenance for 4.9+ to make the requisite changes. It likely would be a source for cherrypick failures when backporting bug fixes, but the surface area would be low. I'm not super worried about breaking other clients that depend on marketplace at this point either.